routes/toSwipe.js

```diff
@@ -27,13 +26,27 @@ router.get('/', ensureAuthenticated, (req, res) => {
 
       // fetch all movies in lists by active filter as well as default filters
       MovieList.find(
-        { $or: [{ filterName: { $in: filters } }, { filterType: 'default' }] },
+        { filterName: { $in: genreFilters } },
         async (error, doc) => {
           try {
             if (err) throw err;
             if (!doc) return [];
             if (doc) {
-              const moviesList = doc[0].films;
+              if (doc.length === 0) {
+                return [];
+              }
+              const tempMoviesList = doc[0].films;
+              const years = timeFilters.map(
+                (x) => parseInt(x.slice(0, -1)) / 10
+              );
+              const moviesList =
+                years.length > 0
+                  ? tempMoviesList.filter(
+                      (x) =>
+                        years.indexOf(Math.floor(parseInt(x['year']) / 10)) !==
+                        -1
+                    )
+                  : tempMoviesList;
```

Comment:
This is more like a suggestion so you don't need to do but:
instead of defining an aux const variable `tempMoviesList` and then create the final `moviesList` from it, why not just create moviesList as let and then conditionally filter it eg:

``` js
let moviesList = doc[0].films; // instead of: const tempMoviesList = doc[0].films;
// code ...
if (years.length) {
 moviesList = moviesList.filter(...)
}
```
This really isn't a big deal but just a thought because it reduces the `const x = a ? b : c` which is mostly good but can be unnecessary

Reply:
cool, thanks for the feedback :)